### PR TITLE
Rename account

### PR DESCRIPTION
**Pull request by aricart**
Added an internal command for renaming an account.

### CODE REVIEW

File: cmd/push_test.go [hunk 1]
```diff
@@ -16,10 +16,11 @@
 package cmd
 
 import (
-	"github.com/nats-io/jwt"
 	"runtime"
 	"testing"
 
+	"github.com/nats-io/jwt"
+
```
Comment (kozlovic) [inline cmd/push_test.go]: remove line here.

File: cmd/rename.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2018-2020 The NATS Authors
```
Comment (kozlovic) [inline cmd/rename.go:2]: Date

File: cmd/rename.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !params.yes {
+				conf := GetConfig()
+				stores := AbbrevHomePaths(conf.StoreRoot)
+				keys := AbbrevHomePaths(store.GetKeysDir())
+				cmd.Printf(`This command makes destructive changes to files and keys. The account 
+rename operation requires that the JWT be reissued with a new name, 
+reusing it's previous ID. This command will move the account, and moves
+users and associated credential files under the new account name. 
+Please the following directories:
```
Comment (kozlovic) [inline cmd/rename.go:56]: "backup" missing?

File: cmd/rename.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+		r.AddWarning("error deleting user %q from account %q", u, p.from)
+	}
+	return r, err
+}
+
+func (p *RenameAccountParams) moveCreds(ctx ActionCtx) (store.Status, error) {
+	r := store.NewDetailedReport(false)
+	fp := ctx.StoreCtx().KeyStore.CalcAccountCredsDir(p.from)
+	if _, err := os.Stat(fp); os.IsNotExist(err) {
+		r.AddOK("creds directory for %q doesn't exist", p.from)
```
Comment (kozlovic) [inline cmd/rename.go:196]: AddError instead?

File: cmd/rename.go [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+func (p *RenameAccountParams) Run(ctx ActionCtx) (store.Status, error) {
+	r := store.NewDetailedReport(true)
+	r.ReportSum = false
+
+	sr, err := p.copyAccount(ctx)
+	r.Add(sr)
+	if err != nil {
+		return r, err
+	}
+	cus, err := p.copyUsers(ctx)
```
Comment (kozlovic) [inline cmd/rename.go:217]: Err is not checked.

File: cmd/rename.go [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+	r.Add(cus)
+
+	mcr, err := p.moveCreds(ctx)
+	r.Add(mcr)
+	if err != nil {
```
Comment (kozlovic) [inline cmd/rename.go:222]: Not needed since your then `return r, err`

File: cmd/rename_test.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018-2020 The NATS Authors
```
Comment (kozlovic) [inline cmd/rename_test.go:2]: Date

File: cmd/rename.go (hunk 4 above)
Comment (aricart) [reply cmd/rename.go:196]: nope - you may not have users.

File: cmd/rename.go (hunk 3 above)
Comment (aricart) [reply cmd/rename.go]: yes.

**Pull request merged**